### PR TITLE
Simplify view-book makefile target.

### PR DESCRIPTION
**Pull request by apogeeoak**
Utilize the functionality of mdbook to open the Theseus book. The make dependency on book can be removed as the book is rebuild before opening. The check for a mdbook install would also be removed in this case. A somewhat useful error is still printed if mdbook cannot be found.

The `build` command could also be replaced with a `watch` command which rebuilds the book whenever the source changes.

```shell
@(&> /dev/null mdbook watch --open $(BOOK_SRC) -d $(BOOK_OUT) &)
```

The method used to suppress the output may not be portable.

### CODE REVIEW

**Comment by kevinaboos** [comment]
This is a great idea, I'd love to avoid dealing with platform-specific things like `wslview` or `xdg-open`. 

Unfortunately, `mdbook build --open` doesn't work for me on WSL or WSL2, which is important because several other folks working on Theseus do use WSL. Perhaps there's some environment variable to set when using `mdbook build --open` in order to allow it to work properly on WSL, but I'm not sure what the solution is. 

It appears to use the `open` crate; that functionality was added to `mdBook` in this commit: https://github.com/rust-lang/mdBook/commit/3ffd24df631fba46fefe750719732624e3b8c4d2

If we can get it to work on WSL (and ideally macOS too), then I'm 100% in favor of merging this in. Also, you gave me the idea to check whether rustdoc supports this, and they do too using `cargo doc --open`.  Unfortunately, like `mdbook build --open`, the cargo doc one also doesn't work on WSL, at least for me.

**Comment by kevinaboos** [comment]
Looking at the source code of the `open` crate, the version used by mdBook is 1.1, which doesn't support WSL. The newer version of `open` 1.7 _does_ support WSL using the `wslview` executable, so once mdBook is updated to use that, then we can merge this in.

**Comment by apogeeoak** [comment]
I was unaware that WSL was not supported. I will try to submit a request to mdBook or rustdoc to update.

**Comment by kevinaboos** [comment]
Looks like the necessary PR was merged into mdBook last week: https://github.com/rust-lang/mdBook/pull/1643

Pending testing on WSL, I think we can unblock this. :+1:

**Comment by apogeeoak** [comment]
If mdBook is installed or updated through `cargo install mdbook` then `mdbook build --open` works on WSL. The latest binary release is not yet updated with the new functionality.

`cargo doc --open` now works under the latest stable compiler version.

**Comment by kevinaboos** [comment]
> If mdBook is installed or updated through `cargo install mdbook` then `mdbook build --open` works on WSL. The latest binary release is not yet updated with the new functionality.
> 
> `cargo doc --open` now works under the latest stable compiler version.

Great to hear! I think I can unblock this now thanks to your confirmation. Also thanks a lot for pushing this through the mdBook repo too!

**Comment by kevinaboos** [comment]
Will test soon on my WSL setup and then merge if everything works.

**Comment by kevinaboos** [comment]
This does work on WSL if we install mdbook from their git repo, but not if we just install it from crates.io, which makes sense because mdbook hasn't been published to crates.io since August.

I'm going to re-mark this as blocked until mdbook publishes an update to crates.io.

**Comment by apogeeoak** [comment]
If you did not want to wait for a new `mdbook` release, the `open` crate provides a small binary with the open functionality. The `open` crate is on a quick release cycle and it could also be used for opening the source documentation. However, this would add another dependency.

**Comment by kevinaboos** [comment]
That's true, however, using the `open` crate isn't as much of an improvement (if any) over the current direct call to `wslview` as using a built-in `--open` flag in `mdBook` or `cargo doc`. No rush, we can wait on this until things stabilize.

**Comment by kevinaboos** [comment]
Oh, funnily enough it looks like mdBook was just updated: https://github.com/rust-lang/mdBook/commit/3e91f9cd5d5e3104e61cbc82f2a18ab9d63eecab

I'll unblock this and test again soon.

**Comment by kevinaboos** [comment]
Tested and working with the latest mdbook version. Thanks again for the contribution, @apogeeoak!

**Review by kevinaboos — APPROVED** [review]

**Comment by apogeeoak** [comment]
You are welcome.

**Pull request merged**
